app/scripts/metamask-controller.js

```diff
@@ -1303,8 +1306,9 @@ export default class MetamaskController extends EventEmitter {
             ? accountTokens[address][chainId].filter(
                 ({ address: tokenAddress }) => {
                   const checksumAddress = toChecksumHexAddress(tokenAddress);
-                  return contractMap[checksumAddress]
-                    ? contractMap[checksumAddress].erc20
+                  return this.preferencesController.store.getState()
+                    .useStaticTokenList && tokenList[checksumAddress]
+                    ? tokenList[checksumAddress].erc20
```

Comment:
I'm not understanding why we check the useStaticTokenList preference here?

Reply:
At the moment, I am also wondering if we need filtering for erc20 at this level with the current tokenList changes because we are filtering out the tokens that are not erc20 in TokenListController itself. 
@danjm Can I get your input on if we can skip the filtering and directly use `accountTokens[address][chainId]`??!!

Reply:
Removed the filtering as the` tokenList` from the controller holds only erc20 tokens.